esp-hal/src/uart.rs

```diff
@@ -329,6 +349,20 @@ impl Default for RxConfig {
 }
 
 impl Config {
+    /// Set the baudrate tolerance of the UART configuration.
+    #[instability::unstable]
+    pub fn with_baudrate_tolerance(mut self, tolerance: BaudrateTolerance) -> Self {
+        self.baudrate_tolerance = match tolerance {
+            BaudrateTolerance::Exact => BaudrateTolerance::Exact,
+            BaudrateTolerance::Closest => BaudrateTolerance::Closest,
+            BaudrateTolerance::ErrorPercent(percentage) => {
+                assert!(percentage > 0 && percentage <= 100);
+                BaudrateTolerance::ErrorPercent(percentage)
+            }
+        };
```

Comment:
```suggestion
        if let BaudrateTolerance::ErrorPercent(percentage) = tolerance {
            assert!(percentage > 0 && percentage <= 100);
        }
        self.baudrate_tolerance = tolerance;
```

Any reason not to simplify this?